fem/kernels.hpp

```diff
@@ -51,192 +95,996 @@ MFEM_HOST_DEVICE inline void LoadBG(const int D1D, const int Q1D,
    MFEM_SYNC_THREAD;
 }
 
+/// Load Bt1d & Gt1d matrices into shared memory
+template<int MD1, int MQ1>
+MFEM_HOST_DEVICE inline void LoadBGt(const int D1D, const int Q1D,
+                                     const ConstDeviceMatrix &b,
+                                     const ConstDeviceMatrix &g,
+                                     double (&sBG)[2][MQ1*MD1])
+{
+   const int tidz = MFEM_THREAD_ID(z);
+   DeviceMatrix Bt(sBG[0], MQ1, MD1);
+   DeviceMatrix Gt(sBG[1], MQ1, MD1);
```

Comment:
Is it a shared memory optimization, otherwise:
```suggestion
   DeviceMatrix Bt(sBG[0], Q1D, D1D);
   DeviceMatrix Gt(sBG[1], Q1D, D1D);
```

Reply:
`done`